packages/language-clojure/lib/main.js

```diff
@@ -1,39 +1,50 @@
+const path = require('path');
+
 exports.activate = function() {
   if (!atom.grammars.addInjectionPoint) return;
 
+  const notDisChild = (node) => {
+    let parent = node.parent
+    while(parent) {
+      if(parent.type === 'dis_expr') return null
+      parent = parent.parent
+    }
+    return node
+  }
+
   atom.grammars.addInjectionPoint('source.clojure', {
     type: 'quoting_lit',
-    language: () => 'source-edn',
-    content: (node) => {
-      let parent = node.parent
-      while(parent) {
-        if(parent.type === 'dis_expr') return null
-        parent = parent.parent
-      }
-      return node
-    },
+    language: () => 'source-clojure-edn',
+    content: notDisChild,
     includeChildren: true,
-    languageScope: 'source.edn',
+    languageScope: 'source.clojure',
     coverShallowerScopes: true
   });
 
   atom.grammars.addInjectionPoint('source.clojure', {
-    type: 'syn_quoting_lit',
-    language: () => 'source-quoted-clojure',
-    content: (node) => node,
+    type: 'source',
+    language: () => 'source-clojure-edn',
+    content: (node, buffer) => {
+      if(path.extname(buffer.getPath()) === '.edn') {
```

Comment:
```suggestion
      if(path.extname(buffer.getPath() ?? '') === '.edn') {
```

The docs say that `path.extname` throws if the given parameter is not a string, and `buffer.getPath()` will return `undefined` if the file is untitled and unsaved.